flexget/plugins/list/series_list.py

```diff
@@ -0,0 +1,438 @@
+        db_series.special_ids = [SeriesListSpecialID(id_name=name) for name in entry.get('special_ids', [])]
+        db_series.prefer_specials = entry.get('prefer_specials')
+        db_series.assume_special = entry.get('assume_special')
+        db_series.tracking = entry.get('tracking')
+
+        # Get list of supported identifiers
+        for id_name in SUPPORTED_IDS:
+            value = entry.get(id_name)
+            if value:
+                log.debug('found supported ID %s with value %s in entry, adding to series', id_name, value)
+                db_series.ids.append(SeriesListID(id_name=id_name, id_value=value))
+        log.debug('adding entry %s', entry)
+        db_list.series.append(db_series)
+        session.commit()
+        return db_series.to_entry()
+
+    @with_session
+    def discard(self, entry, session=None):
+        db_series = self._find_entry(entry, session=session)
+        if db_series:
+            log.debug('deleting series %s', db_series)
+            session.delete(db_series)
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    @with_session
+    def _find_entry(self, entry, session=None):
+        """Finds `SeriesListSeries` corresponding to this entry, if it exists."""
+        for id_name in SUPPORTED_IDS:
+            if entry.get(id_name):
+                log.debug('trying to match series based off id %s: %s', id_name, entry[id_name])
+                res = (self._db_list(session).series.join(SeriesListSeries.ids).filter(
+                    and_(
+                        SeriesListID.id_name == id_name,
+                        SeriesListID.id_value == entry[id_name]))
+                       .first())
+                if res:
+                    log.debug('found series %s', res)
+                    return res
+        # Fall back to title
+        name = entry.get('series_name')
+        if not name:
+            log.verbose('no series name to match, skipping')
+            return
+        log.debug('trying to match series based of name: %s ', name)
+        res = (self._db_list(session).series.filter(SeriesListSeries.title == name.lower()).first())
+        if res:
+            log.debug('found series %s', res)
+        return res
+
+    @property
+    def immutable(self):
+        return False
+
+    @property
+    def online(self):
+        """ Set the online status of the plugin, online plugin should be treated differently in certain situations,
+        like test mode"""
+        return False
+
+    @with_session
+    def get(self, entry, session):
+        match = self._find_entry(entry=entry, session=session)
+        return match.to_entry() if match else None
+
+
+class PluginSeriesList(object):
+    schema = {'type': 'string'}
+
+    @staticmethod
+    def get_list(config):
+        return SeriesList(config)
+
+    def on_task_input(self, task, config):
+        return list(SeriesList(config))
+
+
+@event('plugin.register')
```

Comment:
After this are API helper methods. I agree this isn't the prettiest format, just retained the structure we set with other plugins.